website/docs/r/nsxv_firewall.markdown

```diff
@@ -0,0 +1,255 @@
+  org          = "my-org"
+  vdc          = "my-vdc"
+  edge_gateway = "my-edge-gateway"
+
+  source {
+    ip_addresses       = ["any"]
+    gateway_interfaces = ["internal"]
+  }
+
+  destination {
+    ip_addresses = ["192.168.1.110"]
+  }
+
+  service {
+    protocol = "icmp"
+  }
+
+  service {
+    protocol = "tcp"
+    port     = "443"
+  }
+}
+```
+
+## Example Usage 3 (Use exclusion in source)
+
+```hcl
+resource "vcd_nsxv_firewall" "my-rule-1" {
+  org          = "my-org"
+  vdc          = "my-vdc"
+  edge_gateway = "my-edge-gateway"
+
+  source {
+    exclude            = true
+    gateway_interfaces = ["internal"]
+  }
+
+  destination {
+    ip_addresses = ["any"]
+  }
+
+  service {
+    protocol = "icmp"
+  }
+}
+```
+
+## Example Usage 4 (Deny rule using exclusion and priority set)
+
+```hcl
+resource "vcd_nsxv_firewall" "my-rule-1" {
+  org          = "my-org"
+  vdc          = "my-vdc"
+  edge_gateway = "my-edge-gateway"
+
+  logging_enabled = "true"
+  action          = "deny"
+
+  source {
+    ip_addresses = ["30.10.10.0/24", "31.10.10.0/24"]
+    org_networks = ["org-net-1", "org-net-2"]
+  }
+
+  destination {
+    ip_addresses = ["any"]
+  }
+
+  service {
+    protocol = "icmp"
+  }
+}
+
+resource "vcd_nsxv_firewall" "my-rule-2" {
+  org          = "my-org"
+  vdc          = "my-vdc"
+  edge_gateway = "my-edge-gateway"
+
+  # This attribute allows to ensure rule is inserted above the referred one
+  # in rule processing engine
+  above_rule_id = "${vcd_nsxv_firewall.my-rule-1.id}"
+  name          = "my-friendly-name"
+
+  source {
+    ip_addresses = ["30.10.10.0/24", "31.10.10.0/24"]
+    org_networks = ["org-net-1", "org-net-2"]
+  }
+
+  destination {
+    ip_addresses = ["any"]
+  }
+
+  service {
+    protocol = "icmp"
+  }
+}
+```
+
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+when connected as sysadmin working across different organisations.
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level.
+* `edge_gateway` - (Required) The name of the edge gateway on which to apply the firewall rule.
+* `action` - (Optional) Defines if the rule is set to `accept` or `deny` traffic. Default `accept`
+* `enabled` - (Optional) Defines if the rule is enabaled. Default `true`.
+* `logging_enabled` - (Optional) Defines if the logging for this rule is enabaled. Default `false`.
+* `name` - (Optional) Free text name. Can be duplicate.
+* `rule_tag` - (Optional) This can be used to specify user-controlled rule tag. If not specified,
+it will report rule ID after creation. Must be between 65537-131072.
+* `above_rule_id` - (Optional) This can be used to alter default rule placement order. By default
+every rule is appended to the end of firewall rule list. When a value of another rule is set - this
+rule will be placed above the specified rule.
```

Comment:
You have good examples there, but we also need one with **more than one** rule where `above_rule_id` is used.

Reply:
The example 4 actually uses the `above_rule_id`. I updated its title to explicitly state that.